### PR TITLE
Backstage Refactor

### PR DESCRIPTION
**Pull request by luisfelipefurlan**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
dojot/dojot#2331 dojot/dojot#2301 dojot/dojot#2342

* **Other information**:

### CODE REVIEW

**Review by LuanEdCosta — APPROVED** [review]

File: DockerFile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:8-alpine
+FROM node:12-alpine
```
Comment (cfrancisco) [inline DockerFile:1]: Nice!

File: src/operations/configuration/query.getConfig.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  try {
+    const result = await userPool.query(query);
+    if (result.rowCount) {
+      return (JSON.stringify(result.rows[0].configuration));
+    }
+
+    throw `Could not retrieve configuration from user ${params.user} in tenant ${params.tenant}`;
+  } catch (error) {
+    LOG.error(error);
+    return 'Could not complete operation';
```
Comment (cfrancisco) [inline src/operations/configuration/query.getConfig.js:30]: Should we also throw the error (as in the line above)?

File: src/operations/device/Resolvers.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const getDeviceById = require('./query.getDeviceById')
```
Comment (cfrancisco) [inline src/operations/device/Resolvers.js:1]: Clean isolation, GJ!

File: src/services/service.device.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    const promise = axios.get(`${baseURL}/device/${deviceId}`, getHeader(token)).then((response) => {
+      if (!!response.data) {
+        const {data: {attrs, created, id, label, templates}} = response;
+        values[id] = {attrs, created, id, label, templates, templateID: null}
+      }
+    }).catch(() => Promise.resolve(null));
+    promises.push(promise);
+  })
+
+  await (Promise.all(promises));
```
Comment (cfrancisco) [inline src/services/service.device.js:26]: It sounds a little strange to use the "values" variable as a closure for the callback, but It's okay.
I think these could be done with only one array (returning to values[id] and after Promises.all(values[id]) );

**Comment by cfrancisco** [comment]
Still has one unit test broken :\. For Deep clone, I would suggest don't retrieve the certificate from device object (and adding it as a comment to be updated when the feature arrives).

**Review by cfrancisco — APPROVED** [review]

**Review by LuanEdCosta — APPROVED** [review]

**Pull request merged**